src/mtev_reverse_socket.c

```diff
@@ -287,71 +308,75 @@ command_out(reverse_socket_t *rc, uint16_t id, const char *command) {
 static void
 mtev_reverse_socket_channel_shutdown(reverse_socket_t *rc, uint16_t i, eventer_t e) {
   eventer_t ce = NULL;
-  if(rc->channels[i].pair[0] >= 0)
+  int need_deref = 0;
+  if(rc->data.channels[i].pair[0] >= 0) {
     mtevL(nldeb, "mtev_reverse_socket_channel_shutdown(%s, %d)\n", rc->id, i);
+    need_deref = 1;
+  }
   pthread_mutex_lock(&rc->lock);
-  if(rc->channels[i].pair[0] >= 0) {
-    int fd = rc->channels[i].pair[0];
+  if(rc->data.channels[i].pair[0] >= 0) {
+    int fd = rc->data.channels[i].pair[0];
     eventer_t ce = eventer_find_fd(fd);
```

Comment:
Yeah, that seems broken. Not directly related to the PR at hand, but it's something worth looking into. I'll investigate it.